File: notes-backend/src/main/java/com/notesapp/notes_backend/service/NoteService.java
package com.notesapp.notes_backend.service;

import com.notesapp.notes_backend.dto.request.NoteRequestDto;
import com.notesapp.notes_backend.dto.response.NoteResponseDto;
import com.notesapp.notes_backend.dto.response.PaginatedResponseDto;

import java.util.List;

public interface NoteService {

    NoteResponseDto createNote(NoteRequestDto dto);

    PaginatedResponseDto getAllNotes(
            int pageNo,
            int pageSize,
            String sortBy,
            String sortDir
    );

    NoteResponseDto getNoteById(Long id);

    NoteResponseDto updateNote(Long id, NoteRequestDto dto);

    void deleteNote(Long id);

    List<NoteResponseDto> searchNotes(String keyword);

    List<NoteResponseDto> getPinnedNotes();
}

File: notes-frontend/src/services/noteService.js
import axiosInstance from "../api/axiosInstance";

export const getAllNotes = async (page = 0) => {

  const response = await axiosInstance.get(
    `/notes?pageNo=${page}&pageSize=5`
  );

  return response.data;
};

export const createNote = async (noteData) => {

  const response = await axiosInstance.post(
    "/notes",
    noteData
  );

  return response.data;
};

export const deleteNote = async (id) => {

  const response = await axiosInstance.delete(
    `/notes/${id}`
  );

  return response.data;
};

export const updateNote = async (id, noteData) => {

  const response = await axiosInstance.put(
    `/notes/${id}`,
    noteData
  );

  return response.data;
};

export const searchNotes = async (keyword) => {

  const response = await axiosInstance.get(
    `/notes/search?keyword=${keyword}`
  );

  return response.data;
};